### PR TITLE
implement sort from scratch

### LINKED ISSUE #1: Implement sort(1) from scratch in Go
* [x] Figure out what the command-line tool `sort` does w/o any parameters given
* [x] Implement that behavior as a command-line tool in Go (implement the sort algo yourself)

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #1

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**_Test protocol:_**

**toSort.txt:**
d
h
z
r
2
10
5
78
5
4
2
h
b
u
r
d

**Terminal result for the stock command line command sort (1):**
10
2
2
4
5
5
78
b
d
d
h
h
r
r
u
z

**Code of sort command in GO:**

```
package main

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"os"
)

func main() {
	data, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		fmt.Println(err)
		return
	}

	dataSplit := bytes.Split(data, []byte("\n"))
	lengthOfData := len(dataSplit)

	for i := 0; i < lengthOfData; i++ {
		j := i
		for j > 0 && bytes.Compare(dataSplit[j], dataSplit[j-1]) < 0 {
			dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
			j -= 1
		}
	}
	fmt.Printf("%s", dataSplit)
}
```

**Terminal result for from scratch in GO implemented sort command (2):**
[10 2 2 4 5 5 78 b d d h h r r u z]

**Conclusion:**
The result of both variants are the same. Everything matches and is ordered equally. Both results only differ in the superficial representation. In Terminal result **(1)** our result is printed from top to bottom and every character is printed in a separate/new line. In Terminal result **(2)** our result is printed in one line and with brackets **_[ ]_** at the beginning and end.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Change that "only thing that is different is the layout".

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Change how the code works, not how you describe it.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol. Long explanations aren't needed.

**Comment by justinmueller05082000** [comment]
Test protocol (2):

**toSort.txt:**
```
d
h
z
r
2
10
5
78
5
4
2
h
b
u
r
d
```

**Terminal result for the stock command line command sort (1):**
10
2
2
4
5
5
78
b
d
d
h
h
r
r
u
z

**Terminal result for _changed_ from scratch in GO implemented sort command (2):**
10
2
2
4
5
5
78
b
d
d
h
h
r
r
u
z

**Conclusion:**
Both results are identical.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Your implementation behaves differently on input w/o a final \n. Fix it.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Now the behavior differs for regular files. Fix it.

File: main.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ func main() {
 		return
 	}
 
+	if string(data) == "" {
+		return
+	}
+
```
Comment (Al2Klimov) [inline main.go:21]: That's a special case where the file is empty. I complained about the more general case where there's just no final \n:

```
➜  trainee-sort git:(feature/implement-sort-from-scratch-1) sort <<<'LOLCAT'
LOLCAT
➜  trainee-sort git:(feature/implement-sort-from-scratch-1) go run . <<<'LOLCAT'

LOLCAT
➜  trainee-sort git:(feature/implement-sort-from-scratch-1)
```

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (justinmueller05082000) [reply main.go:21]: So i don't want to print empty lines that are e.g. at the beginning at the end or somewhere in the file? So i just take out those and print afterwards. Should i undo these changes?

Comment (Al2Klimov) [reply main.go:21]: No, it's just about the final \n:

```
➜  trainee-sort git:(feature/implement-sort-from-scratch-1) sort <<<'a

b'

a
b
➜  trainee-sort git:(feature/implement-sort-from-scratch-1) go run . <<<'a

b'


a
b
➜  trainee-sort git:(feature/implement-sort-from-scratch-1)
```

File: main.go [hunk 2]
```diff
@@ -29,5 +30,5 @@ func main() {
 		}
 	}
 	dataJoin := bytes.Join(dataSplit, []byte("\n"))
-	fmt.Printf("%s\n", dataJoin)
+	fmt.Printf("%s\n", strings.TrimPrefix(string(dataJoin), "\n"))
```
Comment (Al2Klimov) [inline main.go:33]: `sort < /dev/null` still doesn't work.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)
* [x] git commit --amend --no-edit
* [x] git push -f origin feature/implement-sort-from-scratch-1

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**